uwblueprint/llsc
# LLSC-24: Scheduling API

## Notion ticket link

[LLSC-24 Scheduling API](https://www.notion.so/uwblueprintexecs/Scheduling-API-11210f3fb1dc80e3bf70da2ad080b838?pvs=25)



## Implementation description
* Created a model for Schedules and TimeBlocks
* A Schedule is composed of several TimeBlocks, which determine ranges in Time
* create_schedule() takes in several time blocks, then creates a Schedule with default values + creates the time_blocks. Uses the ORM relationship mapping to automatically map the time_blocks to the respective schedule id



## Steps to test
1. Make a request to http://localhost:8000/schedules
2. Check FastAPI docs for example input
3. Verify that a new Schedule was created, and that the new Time Blocks have that Schedule id as a field



## What should reviewers focus on?
* If model and schema definitions are sensible


## Checklist
